starky/src/prover.rs

```diff
@@ -164,34 +162,44 @@ where
 {
     let degree = 1 << degree_bits;
 
+    let max_degree_bits = log2_ceil(stark.constraint_degree() - 1);
+    assert!(
+        max_degree_bits <= rate_bits,
+        "Having constraints of degree higher than the rate is not supported yet."
+    );
+    let step = 1 << (rate_bits - max_degree_bits);
+    // When opening the `Z`s polys at the "next" point, need to look at the point `next_step` steps away.
+    let next_step = 1 << max_degree_bits;
```

Comment:
nit: maybe `step_size`? `next_step` kinda sounds like the absolute index vs the delta